### ISSUE TITLE
Featured hashtags are truncated to three no matter the actual number and hidden on smaller screens

### ISSUE THREAD

**Issue by mxamber** [glitch-soc/mastodon#1987]
### Steps to reproduce the problem

1. Add more than three hashtags to the featured hashtags in your preferences
2. Open your profile in the browser, on a screen of at least 1080p (smaller screens omit the featured hashtags via an `@media` rule)

### Expected behaviour

All featured hashtags should be listed on the profile

### Actual behaviour

Only three at most are shown, smaller screens hide them entirely

### Detailed description

The featured hashtags used to list *all* hashtags instead of only the top three (I myself have five pinned). There is no warning or indication about this behaviour (or about hiding them on smaller screens) in the "featured hashtags" panel of the preferences, so users might never know that their more-than-three hashtags are never seen by others. Screens below 1080p aren't at all uncommon either (I tested this on a 1366x768 laptop, for instance).

I suppose this is an upstream issue, but between the extensive glitch rewrites to the v4 UI and the upstream maintainer's unwillingness or inability to fix anything, I figured I might as well open it here.

### Specifications

v4.0.1+glitch on eldritch.cafe, tested on a 19200x1200 screen.

**Comment by Plastikmensch** [comment]
featured hashtag are hardlimited to 3 in `/features/account/components/featured_tags.js`.
That limit should probably either be removed or the number of featured tags a user can set should be limited too.

**Comment by mxamber** [comment]
The thing is, until v4, they weren't. I don't know about the code, but until the v4 update, all five hashtags were displayed properly on my profile.

**Comment by Plastikmensch** [comment]
> The thing is, until v4, they weren't. I don't know about the code, but until the v4 update, all five hashtags were displayed properly on my profile.

Yeah, it got introduced upstream by https://github.com/mastodon/mastodon/commit/23d367f544485eaeed888c707012a8282bbb5806

**Comment by Plastikmensch** [comment]
I messed around a bit. I don't see any reason to not increase the limit to 5, as 3 seems to be arbitrary chosen.
Additional improvement would be to add a hint to the settings page saying "Only 5 tags will be shown on your profile" and/or limit the amount of tags which can be added to 5. It seems a bit strange that more tags can be added than are able to be shown.

As to the featured tags disappearing from the profile on smaller screens... I don't really have a good solution for that.
One thing that semi-fixes it on Desktop is to use `orientation: portrait` instead of `max-width` in the `@screen` rule, but it breaks the mobile layout a bit. I think that can be solved by adjusting the values though.

Both of these don't really fix the core issue which is the new design though and I don't know what upstream is planning, if anything at all, to address this.

**Comment by mxamber** [comment]
Re: the layout, since the sidebar menu scrolls anyway, I don't see a reason to hide them at all. And yeah, whichever the limit (I find 5 too few still), harmonising settings and profile display should be the no. 1 priority.
![image](https://user-images.githubusercontent.com/13611718/205360902-a3930e7b-7904-4db8-ac31-49787b6f582b.png)

**Comment by ClearlyClaire** [comment]
This is a design issue we share with upstream, and I think upstream has plans to address it.

**Comment by Plastikmensch** [comment]
> Re: the layout, since the sidebar menu scrolls anyway, I don't see a reason to hide them at all. And yeah, whichever the limit (I find 5 too few still), harmonising settings and profile display should be the no. 1 priority.

Oh, I didn't account for lists, but yeah, it doesn't really make a difference if it scrolls because of lists or because of featured tags.
I'm more or less indifferent about what the limit ends up being, but I think it shouldn't be higher than 8.
I just mentioned 5 because you said you have 5 selected and work around the scrolling now non-issue.


> This is a design issue we share with upstream, and I think upstream has plans to address it.

That's good, but depending on how long upstream will take, I think it's worth it to at least remove the limit.

**Comment by mxamber** [comment]
Oh and I just noticed: featured hashtags aren't displayed at all if you view a profile while you're logged out. And (related), at least on smaller screens (currentlystuck on 1366x768 for a while), featured profiles aren't displayed at all, both logged in and out.

**Comment by ClearlyClaire** [comment]
> Oh and I just noticed: featured hashtags aren't displayed at all if you view a profile while you're logged out.

Hm, I'm unable to reproduce that.

> And (related), at least on smaller screens (currentlystuck on 1366x768 for a while), featured profiles aren't displayed at all, both logged in and out.

Can't reproduce either, but it's not far from the “mobile” view breakpoint, with the mobile view indeed not showing featured hashtags.
